### ISSUE TITLE
lint: elseif should take init stmt into account

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#311]
It's less trivial to rewrite if/else chain if any of "else if" use short init statement.

Example:
```go
		if p.hasValidIdent() {
			ds.Assigns = append(ds.Assigns, p.getAssign(false))
		} else if p.eqlOffs > 0 {
			p.curErr("invalid var name")
		} else if p.tok == _LitWord {
			ds.Assigns = append(ds.Assigns, &Assign{
				Naked: true,
				Name:  p.getLit(),
			})
		} else if w := p.getWord(); w != nil {
			ds.Assigns = append(ds.Assigns, &Assign{
				Naked: true,
				Value: w,
			})
		} else {
			p.followErr(p.pos, ds.Variant.Value, "names or assignments")
		}
```
`w := p.getWord()` makes this code less trivial to rewrite.

Probably should not trigger on such if/else chains.
